docs/user_needs/zephyr_rtos_user.sdoc

```diff
@@ -0,0 +1,87 @@
+[DOCUMENT]
+TITLE: Zephyr RTOS User
+REQ_PREFIX: ZEP-
+
+[GRAMMAR]
+IMPORT_FROM_FILE: user_needs.sgra
+
+[FREETEXT]
+
+TBD
+
+[/FREETEXT]
+
+[USER_NEED]
+UID: ZEP-127
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: OSS License Allows Statically Linked Private Extensions
+STATEMENT: >>>
+As a Product Manufacturer of resource-constrained devices,
+
+I need open-source software to be licensed such that the proprietary code statically linked with the open source software can be kept private.
+<<<
+
+[USER_NEED]
+STATUS: Draft
+TYPE: User Need
+USER: Zephyr RTOS User
+TITLE: Easily Switch Application to different MCU Architecture
+STATEMENT: >>>
+As a Zephyr RTOS user,
+
+I want to be able to easily switch my application to a different MCU architecture (x86, ARM Cortex-M/A, RISCV etc.).
+<<<
+
+[USER_NEED]
+UID: ZEP-128
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: Development Systems Options for Windows, Linux
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need embedded software development system options for Windows and Linux host operating systems.
+<<<
+
+[USER_NEED]
+UID: ZEP-129
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: RTOS with Preemptive Scheduling
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need an RTOS that provides preemptive scheduling of threads,
+
+so that I can interleave the priorities of work in my device to meet latency constraints.
+<<<
+
+[USER_NEED]
+UID: ZEP-130
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: Sources Configurable for Proprietary HW
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need open source software that configurable for reuse on my proprietary hardware.
+<<<
+
+[USER_NEED]
+UID: ZEP-131
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: Sources Reusable in Regulated Devices
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need software that is reusable in regulated devices,
```

Comment:
`Regulated devices` could be expanded on. Do you want to say:

`software that can be used in products which get certified according to the industry standards, such as IEC 61508, ISO 26262, ...` ?

It is just a suggestion. Let's simply make the statement say this more directly.

Reply:
If we are going to be more direct, then we need statements from Product Manufacturers within each regulated domains:

E.g., 
- As an Avionics Product Manufacturer, I want software that can be certified according to DO-178C;
- As a Global Consumer Wireless Product Manufacturer, I need software that can be certified as compliant with the EU RED security standard, so that I can sell product in the EU.

Reply:
As above, there may be two concepts here: Reuse vs certifiable. Certifiable presumes reuse. Do we need to list them separately when certifiable is one goal of the Zephyr Project?

Reply:
The same as for the other comments: an incomplete list could be moved to a COMMENT and expanded on.